generators/ionic/resources/oauth2/cypress/support/oauth2.ts

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable @typescript-eslint/no-namespace, @typescript-eslint/naming-convention */
+// eslint-disable-next-line spaced-comment
+/// <reference types="cypress" />
+import { apiHost } from './config';
+import { environment } from '../../src/environments/environment';
+
+const {
+  oidcConfig: { redirect_url: redirect_uri, scopes: scope, audience },
+} = environment;
+
+// Get oauth2 basic data
+const getOauth2Data = () =>
+  cy
+    .request({
+      url: `${apiHost}/api/auth-info`,
+      followRedirect: false,
+    })
+    .then(({ body: info }) => {
+      const { issuer, clientId } = info;
+      return cy
+        .request({
+          url: `${issuer}/.well-known/openid-configuration`,
+          followRedirect: false,
+        })
+        .then(({ body: configuration }) => ({
+          info,
+          configuration,
+          qs: {
+            redirect_uri,
+            client_id: clientId,
```

Comment:
This has to be hard-coded to the client ID for the native app on Auth0 and Okta. This is a different between these providers and Keycloak. You also have to hard-code the client ID in `auth-config.service.ts`. https://www.jhipster.tech/security/#update-your-ionic-app